src/main/java/me/i509/fabric/bulkyshulkies/extension/QuickShulkerHook.java

```diff
@@ -0,0 +1,81 @@
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.PacketByteBuf;
+
+public class QuickShulkerHook implements RegisterQuickShulker {
+   public static final Identifier QS$SHULKER_SCROLLABLE_CONTAINER = BulkyShulkies.id("qs_shulker_scrollable_container");
+   public static final Identifier QS$SHULKER_9x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_9x7");
+   public static final Identifier QS$SHULKER_11x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_11x7");
+   public static final Identifier QS$SHULKER_13x7_CONTAINER = BulkyShulkies.id("qs_shulker_container_13x7");
+   @Override
+   public void registerProviders() {
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_SCROLLABLE_CONTAINER, QuickShulkerHook::createScrollable);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_9x7_CONTAINER, QuickShulkerHook::create9x7);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_11x7_CONTAINER, QuickShulkerHook::create11x7);
+      ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_13x7_CONTAINER, QuickShulkerHook::create13x7);
+
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_9x7_CONTAINER, Generic9x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_11x7_CONTAINER, Generic11x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_13x7_CONTAINER, Generic13x7Screen::createScreen);
+      ScreenProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_SCROLLABLE_CONTAINER, ScrollableScreen::createScreen);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_SCROLLABLE_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.COPPER_SHULKER_BOX, ShulkerBlocks.IRON_SHULKER_BOX, ShulkerBlocks.SILVER_SHULKER_BOX, ShulkerBlocks.SLAB_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_13x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.OBSIDIAN_SHULKER_BOX, ShulkerBlocks.PLATINUM_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_11x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.DIAMOND_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_9x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.GOLD_SHULKER_BOX);
```

Comment:
Have you tested any of the colored boxes? Cause these only refer to the undyed blocks.

Reply:
They all do correctly work besides the ender slab as I just now realized they exist... whoops
Not too happen with this implementation  on quick shulkers end, but it avoids adding all 136 boxes individually

Reply:
There is a getItemStack method on BasicShulkerBox, which will give you any color of the same type. Just note you need to then copy the BlockEntityTag.